packages/roosterjs-editor-core/lib/undo/Undo.ts

```diff
@@ -128,22 +126,16 @@ export default class Undo implements UndoService {
         return this.getSnapshotsManager().canMove(1 /*nextSnapshot*/);
     }
 
-    /**
-     * Add an undo snapshot
-     */
-    public addUndoSnapshot(): string {
-        let snapshot = this.editor.getContent(
-            false /*triggerExtractContentEvent*/,
-            true /*markSelection*/
-        );
-        this.getSnapshotsManager().addSnapshot(snapshot);
+    public addUndoSnapshot(): UndoSnapshot {
+        const snapshot = this.editor.undoSnapshotTranslator.getUndoSnapshot();
+        this.getSnapshotsManager().addSnapshot(JSON.stringify(snapshot));
         this.hasNewContent = false;
         return snapshot;
     }
 
     protected getSnapshotsManager(): UndoSnapshotsService {
         if (!this.undoSnapshots) {
-            this.undoSnapshots = new UndoSnapshots(this.maxBufferSize);
+            this.undoSnapshots = new UndoSnapshotsService(this.maxBufferSize);
```

Comment:
Isn't UndoSnapshotsService an interface? I didn't see you change it to a class in your change.